routing/bandwidth.go

```diff
@@ -18,7 +22,40 @@ type bandwidthHints interface {
 	// will be used. If the channel is unavailable, a zero amount is
 	// returned.
 	availableChanBandwidth(channelID uint64,
-		amount lnwire.MilliSatoshi) (lnwire.MilliSatoshi, bool)
+		amount lnwire.MilliSatoshi,
+		htlcBlob fn.Option[tlv.Blob]) (lnwire.MilliSatoshi, bool)
+}
+
+// TlvTrafficShaper is an interface that allows the sender to determine if a
+// payment should be carried by a channel based on the TLV records that may be
+// present in the `update_add_htlc` message or the channel commitment itself.
+type TlvTrafficShaper interface {
```

Comment:
I think this belongs to the `link`?

Reply:
No, I don't think so. This clearly affects decisions made by the router. We can discuss the name if you have any ideas. But the tasks for this component are: Determine and choose what outbound channels can be used for a specific outgoing payment, then produce custom records for the first hop of such a payment.